### PR TITLE
[macOS] Retrieve the list of Android packages for installing dynamically

### PR DESCRIPTION
**Pull request by dmitry-shibanov**
# Description
In this pull request we add logic to install android tools dynamic for macOS.

## Check list
- [x] Related issue / work item is attached
- [x] Tests are written (if applicable)
- [x] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

**Review by AlenaSviridenko — CHANGES_REQUESTED** [review]
Looks like Android SDK Platforms are lost in the report and on the image 
image


There are a lot of errors like ones below in the logs. Please take a look.
```
vsphere-clone: /tmp/script_5797.sh: line 13: ((: 11 >= platforms;android-10 : syntax error: invalid arithmetic operator (error token is ";android-10 ")
==> vsphere-clone: /tmp/script_5797.sh: line 13: ((: 12 >= platforms;android-10 : syntax error: invalid arithmetic operator (error token is ";android-10 ")
    vsphere-clone: version is 12
    vsphere-clone: version is 13
==> vsphere-clone: /tmp/script_5797.sh: line 13: ((: 13 >= platforms;android-10 : syntax error: invalid arithmetic operator (error token is ";android-10 ")
    vsphere-clone: version is 14
==> vsphere-clone: /tmp/script_5797.sh: line 13: ((: 14 >= platforms;android-10 : syntax error: invalid arithmetic operator (error token is ";android-10 ")
    vsphere-clone: version is 15
==> vsphere-clone: /tmp/script_5797.sh: line 13: ((: 15 >= platforms;android-10 : syntax error: invalid arithmetic operator (error token is ";android-10 ")
==> vsphere-clone: /tmp/script_5797.sh: line 13: ((: 16 >= platforms;android-10 : syntax error: invalid arithmetic operator (error token is ";android-10 ")
    vsphere-clone: version is 16
```

Also, I think we don't need to output version if we are not installing it.

File: images/macos/provision/core/android-toolsets.sh [hunk 1]
```diff
@@ -48,17 +66,14 @@ echo y | $SDKMANAGER "cmake;3.6.4111459"
 echo "Installing latest ndk..."
 echo y | $SDKMANAGER "ndk-bundle"
 
-for platform_name in "${ANDROID_PLATFORM_LIST[@]}"
-do
-    echo "Installing platform $platform_name ..."
-    echo y | $SDKMANAGER "platforms;$platform_name"
-done
+availablePlatforms=($(${ANDROID_HOME}/tools/bin/sdkmanager --list | sed -n '/Available Packages:/,/^$/p' | grep "platforms;android-" | cut -d"|" -f 1))
```
Comment (al-cheb) [inline images/macos/provision/core/android-toolsets.sh:69]: Use the same approach as you use for allBuildTools without sed

**Review by al-cheb — APPROVED** [review]

File: images/macos/software-report/SoftwareReport.Android.psm1 [hunk 2]
```diff
@@ -28,6 +28,11 @@ function Get-AndroidInstalledPackages {
     return $androidInstalledPackages
 }
 
+function Get-AndroidPackages {
```
Comment (al-cheb) [inline images/macos/software-report/SoftwareReport.Android.psm1:31]: Something goes wrong:

```
 vsphere-clone: Discovering in /Users/runner/image-generation/tests/Android.Tests.ps1.
    vsphere-clone: WARNING: The names of some imported commands from the module 'Tests.Helpers' include unapproved verbs that might make them less discoverable. To find the commands with unapproved verbs, run the Import-Module command again with the Verbose parameter. For a list of approved verbs, type Get-Verb.
    vsphere-clone: System.Management.Automation.CommandNotFoundException: The term 'Get-AndroidPackages' is not recognized as the name of a cmdlet, function, script file, or operable program.
    vsphere-clone: Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
    vsphere-clone:    at System.Management.Automation.ExceptionHandlingOps.CheckActionPreference(FunctionContext funcContext, Exception exception)
    vsphere-clone:    at System.Management.Automation.Interpreter.ActionCallInstruction`2.Run(InterpretedFrame frame)
    vsphere-clone:    at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
    vsphere-clone:    at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
    vsphere-clone:    at System.Management.Automation.Interpreter.Interpreter.Run(InterpretedFrame frame)
    vsphere-clone:    at System.Management.Automation.Interpreter.LightLambda.RunVoid1[T0](T0 arg0)
    vsphere-clone:    at System.Management.Automation.PSScriptCmdlet.RunClause(Action`1 clause, Object dollarUnderbar, Object inputToProcess)
    vsphere-clone:    at System.Management.Automation.PSScriptCmdlet.DoEndProcessing()
    vsphere-clone:    at System.Management.Automation.CommandProcessorBase.Complete()
    vsphere-clone: at <ScriptBlock>, /Users/runner/image-generation/tests/Android.Tests.ps1: line 8
    vsphere-clone: at New-Block, /usr/local/share/powershell/Modules/Pester/5.0.4/Pester.psm1: line 693
    vsphere-clone: at Describe, /usr/local/share/powershell/Modules/Pester/5.0.4/Pester.psm1: line 8042
    vsphere-clone: at <ScriptBlock>, /Users/runner/image-generation/tests/Android.Tests.ps1: line 6
    vsphere-clone: at <ScriptBlock>, /usr/local/share/powershell/Modules/Pester/5.0.4/Pester.psm1: line 2724
    vsphere-clone: at Invoke-File, /usr/local/share/powershell/Modules/Pester/5.0.4/Pester.psm1: line 2733
    vsphere-clone: at Invoke-BlockContainer, /usr/local/share/powershell/Modules/Pester/5.0.4/Pester.psm1: line 2662
    vsphere-clone: at Discover-Test, /usr/local/share/powershell/Modules/Pester/5.0.4/Pester.psm1: line 1295
    vsphere-clone: at Invoke-Test, /usr/local/share/powershell/Modules/Pester/5.0.4/Pester.psm1: line 2195
    vsphere-clone: at Invoke-Pester<End>, /usr/local/share/powershell/Modules/Pester/5.0.4/Pester.psm1: line 4524
    vsphere-clone: at <ScriptBlock>, /Users/runner/image-generation/tests/Run-Tests.ps1: line 25

```

**Review by al-cheb — CHANGES_REQUESTED** [review]

**Review by AlenaSviridenko — APPROVED** [review]

File: images/macos/software-report/SoftwareReport.Android.psm1 (hunk 2 above)
Comment (dmitry-shibanov) [reply images/macos/software-report/SoftwareReport.Android.psm1:31]: Done.

File: images/macos/provision/core/android-toolsets.sh (hunk 1 above)
Comment (dmitry-shibanov) [reply images/macos/provision/core/android-toolsets.sh]: Done.

**Review by al-cheb — APPROVED** [review]

**Review by maxim-lobanov — APPROVED** [review]

**Pull request merged**